Wikipedia/Code/MWKDataStore.m

```diff
@@ -68,84 +73,95 @@ + (MWKDataStore *)shared {
+        WMFSession *session = [[WMFSession alloc] initWithConfiguration:configuration];
+        session.authenticationDelegate = self;
+        WMFAuthenticationManager *authenticationManager = [[WMFAuthenticationManager alloc] initWithSession:session configuration:configuration];
+        authenticationManager.delegate = self;
+        self.session = session;
+        self.configuration = configuration;
+        self.authenticationManager = authenticationManager;
         self.containerURL = containerURL;
         self.basePath = [self.containerURL URLByAppendingPathComponent:@"Data" isDirectory:YES].path;
         [self setupLegacyDataStore];
-        NSDictionary *infoDictionary = [self loadSharedInfoDictionaryWithContainerURL:containerURL];
-        self.crossProcessNotificationChannelName = infoDictionary[@"CrossProcessNotificiationChannelName"];
-        [self setupCrossProcessCoreDataNotifier];
         [self setupCoreDataStackWithContainerURL:containerURL];
+        [self setupCoreDataSynchronizersWithContainerURL:containerURL];
+        [self startSynchronizingLibraryContexts];
         [self setupHistoryAndSavedPageLists];
         self.languageLinkController = [[MWKLanguageLinkController alloc] initWithManagedObjectContext:self.viewContext];
         self.feedContentController = [[WMFExploreFeedContentController alloc] init];
         self.feedContentController.dataStore = self;
         self.feedContentController.siteURLs = self.languageLinkController.preferredSiteURLs;
         [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didReceiveMemoryWarningWithNotification:) name:UIApplicationDidReceiveMemoryWarningNotification object:nil];
-        self.wikidataDescriptionEditingController = [[WikidataDescriptionEditingController alloc] initWithSession:[WMFSession shared] configuration:[WMFConfiguration current]];
-        self.remoteNotificationsController = [[RemoteNotificationsController alloc] initWithSession:[WMFSession shared] configuration:[WMFConfiguration current] preferredLanguageCodesProvider:self.languageLinkController];
-        WMFArticleSummaryFetcher *fetcher = [[WMFArticleSummaryFetcher alloc] initWithSession:[WMFSession shared] configuration:[WMFConfiguration current]];
+        self.wikidataDescriptionEditingController = [[WikidataDescriptionEditingController alloc] initWithSession:session configuration:configuration];
+        self.remoteNotificationsController = [[RemoteNotificationsController alloc] initWithSession:session configuration:configuration preferredLanguageCodesProvider:self.languageLinkController];
+        WMFArticleSummaryFetcher *fetcher = [[WMFArticleSummaryFetcher alloc] initWithSession:session configuration:configuration];
         self.articleSummaryController = [[WMFArticleSummaryController alloc] initWithFetcher:fetcher dataStore:self];
         self.mobileviewConverter = [[MobileviewToMobileHTMLConverter alloc] init];
```

Comment:
I could be missing this somewhere but I don't see the new `notificationsController` property getting instantiated anywhere in this class.

Reply:
Great catch - it's definitely missing. I'll add it.